newsfragments/4523.added.md

```diff
@@ -0,0 +1 @@
+* Added a Rust wrapper for `PyMutex`, available on Python 3.13 and newer.
```

Comment:
Need to update for `with_critical_section_mutex` (and / or split that into a separate PR).